app/services/procurer/fallback.rb

```diff
@@ -0,0 +1,42 @@
+module Procurer
+  class Fallback < Procurer::Service
+    def initialize(name)
+      # assumed that the supplier will have the same base_url
+      url = "https://5f2be0b4ffc88500167b85a0.mockapi.io/suppliers/#{name}"
```

Comment:
A comment that this might not be the best approach.

Each supplier can have a different URL. That said, that info could be stored at database level as you intended.

Reply:
that's true, ill update so the processor accepts a full URL instead